src/supervised/distance.jl

```diff
@@ -108,6 +108,40 @@ islipschitzcont_deriv(::L2DistLoss) = true
 isconvex(::L2DistLoss) = true
 isstronglyconvex(::L2DistLoss) = true
 
+
+# ===========================================================
+# L(y, t) = 1 - cos((y-t)*2π/c)
+
+"""
+`PeriodicLoss <: DistanceLoss`
+
+Measures distance on a circle of specified circumference.
+""" 
+immutable PeriodicLoss{T<:AbstractFloat} <: DistanceLoss
+    k::T   # k = 2π/circumference
+    PeriodicLoss(circ::T) = new(T(2π/circ))
```

Comment:
Good catch! Not sure what to do about negative circumference - since cosine is an even function it is sort of valid and if you accidentally do `PeriodicLoss(-5)` it will behave the same as `PeriodicLoss(5)`. But from a semantics/mathematical perspective a negative circumference doesn't make sense. I'm going to throw an error, unless you think it is better not to interfere.